woodwork/type_sys/utils.py

```diff
@@ -22,6 +23,12 @@ def col_is_datetime(col, datetime_format=None):
     if len(col) == 0:
         return False
 
+    try:
+        if pd.api.types.is_numeric_dtype(pd.Series(col.values.tolist())):
+            return False
+    except AttributeError:
+        pass
```

Comment:
I was looking (unsuccessfully) for alternatives to this as it seems a little confusing why this is needed. Our future selves might benefit from a comment here explaining why this try/except block is necessary.